presenters/thread.rb

```diff
@@ -76,35 +72,34 @@ def to_hash with_responses=false, resp_skip=0, resp_limit=nil, recursive=true
   #     children, if recursive is true)
   #   response_count
   #     The total number of responses
-  def get_paged_merged_responses(thread_id, responses, skip, limit, recursive=false)
+  def get_paged_merged_responses(thread_id, responses, skip, limit, recursive=true)
     response_ids = responses.only(:_id).sort({"sk" => 1}).to_a.map{|doc| doc["_id"]}
     paged_response_ids = limit.nil? ? response_ids.drop(skip) : response_ids.drop(skip).take(limit)
-    if recursive
-      content = Comment.where(comment_thread_id: thread_id).
-        or({:parent_id => {"$in" => paged_response_ids}}, {:id => {"$in" => paged_response_ids}}).
-        sort({"sk" => 1})
-    else
-      content = Comment.where(comment_thread_id: thread_id, "parent_ids" => []).
-        where({:id => {"$in" => paged_response_ids}}).sort({"sk" => 1})
-    end
-    {"responses" => merge_response_content(content), "response_count" => response_ids.length}
+    content = Comment.where(comment_thread_id: thread_id).
+      or({:parent_id => {"$in" => paged_response_ids}}, {:id => {"$in" => paged_response_ids}}).
+      sort({"sk" => 1})
+    {"responses" => merge_response_content(content, recursive), "response_count" => response_ids.length}
   end
 
   # Takes content output from Mongoid in a depth-first traversal order and
   # returns an array of first-level response hashes with content represented
-  # hierarchically, with a comment's list of children in the key "children".
-  def merge_response_content(content)
+  # hierarchically, with "children_count" as no. of comments to a response
+  # and a comment's list of children in the key "children" if recursive = true
+  def merge_response_content(content, recursive=true)
     top_level = []
     ancestry = []
     content.each do |item|
-      item_hash = item.to_hash.merge("children" => [])
+      item_hash = item.to_hash.merge({"children" => [], "children_count" => 0})
       if item.parent_id.nil?
         top_level << item_hash
         ancestry = [item_hash]
       else
         while ancestry.length > 0 do
           if item.parent_id == ancestry.last["id"]
-            ancestry.last["children"] << item_hash
+            if recursive
+              ancestry.last["children"] << item_hash
+            end
+            ancestry.last["children_count"] += 1
```

Comment:
1. The purpose and name of the parameter "recursive" is confusing.  From this code, it seems like all of this data is already loaded into memory, and the recursive=false isn't doing anything except skipping filling these children arrays.  It doesn't make the method run any more quickly, or have any other advantage that I see.
2. I agree that the children array is being created here by this method.  However, after running this method (as originally written), you would have a children array for each response, and the child_count would just be the length of that array.
Are you trying to shrink the size of the data that is returned?  Other?